code/processes/gateway.q

```diff
@@ -454,8 +454,19 @@ asyncexecjpts:{[query;servertype;joinfunction;postback;timeout;sync]
    if[99h<>type servertype;
      // its a list of servertypes e.g. `rdb`hdb
      servertype:distinct servertype,();
-     missing:servertype except exec distinct servertype from .gw.servers where active;
-     if[count missing; errStr:.gw.errorprefix,"not all of the requested server types are available; missing "," " sv string missing];
+     //list of active servers
```

Comment:
Can this logic be moved to the `getserverids` function?

Reply:
Am planning to put the logic in `getserverids` function for the sync queries (pre 3.6) but yeah might be an idea to have `asyncexecjpts` call that function too rather than duplicating code.